skll/utilities/generate_predictions.py

```diff
@@ -71,18 +83,29 @@ def predict(self, data):
         # compute the predictions from the learner
         preds = self._learner.predict(data)
         preds = preds.tolist()
+        labels = self._learner.label_list
 
+        # Create file header list, and transform predictions as needed
+        # depending on the specified prediction arguments.
         if self._learner.probability:
-            if self.threshold is None:
-                return [pred[self._pos_index] for pred in preds]
+            if self.all_labels:
+                self.output_file_header = ["id"] + [str(x) for x in labels]
+            elif self.threshold is None:
+                label = self._learner.label_dict[self._pos_index]
```

Comment:
This line seems to be a typo error. `self._pos_index` appears to be an index in an array but it is used here as a key in a dictionary. Was this supposed to be `label = self._learner.label_list[self._pos_index]` ?

Reply:
You are right. Since `label_dict` maps string labels to ints, and `self._pos_index` is an int, we should be using `label_list` here not `label_dict`. I think we haven't run into this bug because the labels have also been ints for most of our problems? @aoifecahill @Lguyogiro any thoughts on this?

Reply:
Let me move this into a separate issue. since this PR is actually closed.